src/NHSD.BuyingCatalogue.Solutions.Persistence/NHSD.BuyingCatalogue.Solutions.Persistence.csproj

```diff
@@ -34,5 +34,14 @@
       <LastGenOutput>DocumentsAPI.cs</LastGenOutput>
     </None>
   </ItemGroup>
-  
+
+  <ItemGroup>
+    <AssemblyAttribute Include="System.Runtime.CompilerServices.InternalsVisibleTo">
+      <_Parameter1>$(AssemblyName).UnitTests</_Parameter1>
+    </AssemblyAttribute>
+
+    <AssemblyAttribute Include="System.Runtime.CompilerServices.InternalsVisibleTo">
```

Comment:
Can't remember why we need this? Would you mind giving a brief overview?

Reply:
The `DocumentRepository` class was `public` prior to this commit but it doesn't need to be because the only place it is referenced outside of the Persistence project is in the tests. Therefore it is now `internal` but this obviously requires making `internal` classes available to the test project, hence `InternalsVisibleTo`.

Reply:
My comment here referred more to the massive public key in the project file. I couldn't remember why we need to bring in DynamicProxyGenAssembly2.

Reply:
Ah, that's for Moq.

Reply:
👍